### PR TITLE
add supabase + pgvector datastore provider

### PR DESCRIPTION
**Pull request by egor-romanov**
## What kind of change does this PR introduce?

- add supabase as a datastore provider
- add necessary migrations with
  - Postgres function to search closest vectors using pgvector extension
  - table to store embeddings
  - enable pgvector postgres extension
- update readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -491,6 +491,30 @@ Environment Variables:
...
+| Name             | Required | Description                                  | Default    |
+| ---------------- | -------- | -------------------------------------------- | ---------- |
+| `DATASTORE`      | Yes      | Datastore name, set to `supabase`            |            |
+| `BEARER_TOKEN`   | Yes      | Secret token                                 |            |
+| `OPENAI_API_KEY` | Yes      | OpenAI API key                               |            |
+| `SUPABASE_URL`   | Yes      | Supabase Project URL                         |            |
+| `SUPABASE_KEY`   | Yes      | Supabase Project API key (anon or service)   |            |
+| `INIT_DB`        | Optional | Set to true if you want to run DB migrations | `False`    |
+| `PG_HOST`        | Optional | Postgres host                                | none       |
+| `PG_PORT`        | Optional | Postgres port                                | `6543`     |
```
Comment (J0) [inline README.md]: /nit not sure if we want to add an explanatory note that we're using pg_bouncer to explain why we're using port 6543 instead of 5432 - suspect that we might get some questions

File: README.md [hunk 2]
```diff
@@ -101,9 +101,9 @@ The plugin exposes the following endpoints for upserting, querying, and deleting
...
+The detailed specifications and examples of the request and response models can be found by running the app locally and navigating to <http://0.0.0.0:8000/openapi.json>, or in the OpenAPI schema [here](/.well-known/openapi.yaml). Note that the OpenAPI schema only contains the `/query` endpoint, because that is the only function that ChatGPT needs to access. This way, ChatGPT can use the plugin only to retrieve relevant documents based on natural language queries or needs. However, if developers want to also give ChatGPT the ability to remember things for later, they can use the `/upsert` endpoint to save snippets from the conversation to the vector database. An example of a manifest and OpenAPI schema that give ChatGPT access to the `/upsert` endpoint can be found [here](/examples/memory).
```
Comment (kiwicopple) [inline README.md]: I don't think you should make these sort of changes in this PR. keep it relevant to the task

File: README.md [hunk 3]
```diff
@@ -233,14 +233,14 @@ You can run Weaviate in 4 ways:
...
-      - `OIDC Authentication` – Enabled by default. This requires a username and password to access your instance.
-  4.  Wait for a few minutes until your cluster is ready. You will see a green tick ✔️ when it's done. Copy your cluster URL.
+  1. Navigate to [Weaviate Cloud Console](https://console.weaviate.io/).
+  2. Register or sign in to your WCS account.
+  3. Create a new cluster with the following settings:
+     - `Name` – a unique name for your cluster. The name will become part of the URL used to access this instance.
+     - `Subscription Tier` – Sandbox for a free trial, or contact [hello@weaviate.io](mailto:hello@weaviate.io) for other options.
+     - `Weaviate Version` - The latest version by default.
+     - `OIDC Authentication` – Enabled by default. This requires a username and password to access your instance.
+  4. Wait for a few minutes until your cluster is ready. You will see a green tick ✔️ when it's done. Copy your cluster URL.
```
Comment (kiwicopple) [inline README.md]: Likewise - move this to another PR if you want

File: README.md [hunk 4]
```diff
@@ -491,6 +491,30 @@ Environment Variables:
...
+
+Environment Variables:
+
+| Name             | Required | Description                                  | Default    |
+| ---------------- | -------- | -------------------------------------------- | ---------- |
+| `DATASTORE`      | Yes      | Datastore name, set to `supabase`            |            |
+| `BEARER_TOKEN`   | Yes      | Secret token                                 |            |
+| `OPENAI_API_KEY` | Yes      | OpenAI API key                               |            |
+| `SUPABASE_URL`   | Yes      | Supabase Project URL                         |            |
+| `SUPABASE_KEY`   | Yes      | Supabase Project API key (anon or service)   |            |
```
Comment (kiwicopple) [inline README.md]: should we split this into 2? `SUPABASE_ANON_KEY` and `SUPABASE_SERVICE_ROLE_KEY`. 

we might need both in the future

**Review by kiwicopple — APPROVED** [review]

**Review by J0 — APPROVED** [review]
